### DISCUSSION TITLE
How long will the temporary brownout last?

### DISCUSSION THREAD

**Discussion by egzon-korenica** [community/community#24033] [Other Feature Feedback, Questions, & Ideas]
<p>Today there was a temporary brownout process ongoing for SHA-1 RSA keys thus making it impossible for me to use GitHub on some platforms. However, that does not apply to every repository only to some of them.</p>
<p>Is there any approximate time or date when this brownout process will stop?</p>

**Comment by jsoref** [comment, marked as answer]
<p>Why don’t you just create a new key and add it to your account?</p>
<p>The idea of a brownout is to help users identify systems that need to be updated. You’ve identified things, start updating them.</p>
<p>Eventually your SHA-1 keys will not work at all.</p>
<aside class="onebox allowlistedgeneric" data-onebox-src="https://github.blog/2021-09-01-improving-git-protocol-security-github/">
  <header class="source">
      <img src="https://github.blog/wp-content/uploads/2019/01/cropped-github-favicon-512.png?fit=32%2C32" class="site-icon" width="32" height="32">

      <a href="https://github.blog/2021-09-01-improving-git-protocol-security-github/" target="_blank" rel="noopener nofollow ugc" title="05:00PM - 01 September 2021">The GitHub Blog – 1 Sep 21</a>
  </header>

  <article class="onebox-body">
    <div class="aspect-image" style="--aspect-ratio:690/362;"><img src="https://user-images.githubusercontent.com/2119212/181080878-6227ae23-7fe0-49ae-a34c-ece680967d5c.png" class="thumbnail" width="690" height="362" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/c/2/c25410e336a7f8815e23b481139793f4faef75d1_2_690x362.png, https://aws1.discourse-cdn.com/github/optimized/3X/c/2/c25410e336a7f8815e23b481139793f4faef75d1_2_1035x543.png 1.5x, https://aws1.discourse-cdn.com/github/original/3X/c/2/c25410e336a7f8815e23b481139793f4faef75d1.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/c/2/c25410e336a7f8815e23b481139793f4faef75d1_2_10x10.png"></div>

<h3><a href="https://github.blog/2021-09-01-improving-git-protocol-security-github/" target="_blank" rel="noopener nofollow ugc">Improving Git protocol security on GitHub | The GitHub Blog</a></h3>

  <p>We’re changing which keys are supported in SSH and removing unencrypted Git protocol. If you’re an SSH user, read on for the details and timeline.</p>

  <p>
    <span class="label1">Est. reading time: 6 minutes</span>
  </p>

  </article>

  <div class="onebox-metadata">
    
    
  </div>

  <div style="clear: both"></div>
</aside>

**Comment by egzon-korenica** [comment]
<p>Read the article already and I understand what the solution is but the platform which I use has yet to support SHA-2.</p>
<p>A brownout helps users identify systems that need to be updated but doesn’t mean it’s the final removal/deprecation. That will happen on March 15th.</p>
<p>A brownout doesn’t mean SHA-1 keys aren’t supposed to work anymore, it is supposed to be the case for March 15th</p>

**Comment by jsoref** [comment]
<p>What platform are you on that doesn’t support SHA-2?</p>
<p>If you’re using RHEL6 (or older), you can probably package a newer version of openssh yourself and install that…</p>

**Comment by egzon-korenica** [comment]
<p>A SaaS deployment tool</p>

**Comment by jsoref** [comment]
<p>I’d expect 24 hours.</p>
<p>I hope you’ve already:</p>
<ul>
<li>Filed a ticket against SaaS vendor</li>
<li>Started planning to replace SaaS vendor</li>
</ul>

**Comment by egzon-korenica** [comment]
<p>Thank you for your insight, yes migration is always an option but not something that can be done immediately.</p>

**Comment by mpboom** [comment]
<p>Hi <a class="mention" href="/egzon-korenica">@egzon-korenica</a> ! I’m not sure if there is even a predefined set time for the duration of a brownout, but <a class="mention" href="/jsoref">@jsoref</a> is definitely right. These brownouts are a warning to migrate <em>as soon as possible</em> to a better alternative. If your SaaS provider can’t get this right straight away, I’d reconsider doing business with them. Good luck with fixing it!</p>
